pkg/utils/secrets/certificates.go

```diff
@@ -355,3 +357,45 @@ func GenerateCertificateAuthorities(k8sClusterClient kubernetes.Interface, exist
 
 	return generatedSecrets, certificateAuthorities, nil
 }
+
+// SelfGenerateTLSServerCertificate generates a new CA certificate and signs a server certificate with it. It'll return
+// the certificate objects and store the bytes in a temporary directory whose paths will also be returned.
+func SelfGenerateTLSServerCertificate(name string, dnsNames []string) (*Certificate, string, string, error) {
+	certificateDir, err := ioutil.TempDir("", "self-generated-server-certificates")
```

Comment:
```suggestion
	certificateDir, err := ioutil.TempDir("", "self-generated-server-certificates-")
```

For better readability , currently it is looking like `/tmp/self-generated-server-certificates394803357/` but `/tmp/self-generated-server-certificates-394803357/` is looking better.